$OpenBSD: patch-src_kernel_qdragobject_cpp,v 1.4 2018/04/08 08:36:31 espie Exp $
using shorts for colors is obviously me... (the <<8 is ill-defined)
just do it correctly with ushort.
Index: src/kernel/qdragobject.cpp
--- src/kernel/qdragobject.cpp.orig
+++ src/kernel/qdragobject.cpp
@@ -893,6 +893,16 @@ bool QTextDrag::decode( const QMimeSource* e, QString&
 {
     if(!e)
 	return FALSE;
+        
+    // when subtype is not specified, try text/plain first, otherwise this may read
+    // things like text/x-moz-url even though better targets are available
+    if( subtype.isNull()) {
+        QCString subtmp = "plain";
+        if( decode( e, str, subtmp )) {
+            subtype = subtmp;
+            return true;
+        }
+    }
 
     if ( e->cacheType == QMimeSource::Text ) {
 	str = *e->cache.txt.str;
@@ -1739,9 +1749,9 @@ QColorDrag::QColorDrag( QWidget *dragsource, const cha
 
 void QColorDrag::setColor( const QColor &col )
 {
-    short r = (col.red()   << 8) | col.red();
-    short g = (col.green() << 8) | col.green();
-    short b = (col.blue()  << 8) | col.blue();
+    ushort r = (col.red()   << 8) | col.red();
+    ushort g = (col.green() << 8) | col.green();
+    ushort b = (col.blue()  << 8) | col.blue();
 
     // make sure we transmit data in network order
     r = htons(r);
